### PR TITLE
Add Waymo Dataset

### PR DESCRIPTION
**Pull request by sanskar107**
Add script (`scripts/preprocess_waymo.py`) for converting raw tfrecord files to KITTI format.
Add dataset class for Waymo Dataset.

### CODE REVIEW

File: scripts/preprocess_waymo.py [hunk 1]
```diff
@@ -0,0 +1,450 @@
...
+
+            if not self.is_test:
+                self.save_label(frame, file_idx, frame_idx)
+
+    def __len__(self):
+        return len(self.tfrecord_files)
+
+    def save_image(self, frame, file_idx, frame_idx):
+        self.prefix = ''
+        import cv2
```
Comment (YilingQiao) [inline scripts/preprocess_waymo.py:150]: Is it necessary to add `cv2` as a dependency in our package?  If we only want to use imwrite.

File: scripts/preprocess_waymo.py [hunk 2]
```diff
@@ -0,0 +1,450 @@
...
+            if bounding_box is None or name is None:
+                name = '0'
+                bounding_box = (0, 0, 0, 0)
+
+            my_type = self.type_list[obj.type]
+
+            if my_type not in self.selected_waymo_classes:
+                continue
+
+            # if self.filter_empty_3dboxes and obj.num_lidar_points_in_box < 1:
```
Comment (YilingQiao) [inline scripts/preprocess_waymo.py:252]: Is this test code?

File: scripts/preprocess_waymo.py (hunk 1 above)
Comment (sanskar107) [reply scripts/preprocess_waymo.py]: Done. changed to npy files

File: scripts/preprocess_waymo.py (hunk 2 above)
Comment (sanskar107) [reply scripts/preprocess_waymo.py:253]: Ideally we may want to remove empty box, but I can't find `num_lidar_points_in_box` attribute in `frame` dtype within wamo toolkit. So I left it commented for now.

**Review by Prantl — APPROVED** [review]

File: scripts/preprocess_waymo.py (hunk 2 above)
Comment (tianweiy) [reply scripts/preprocess_waymo.py:253]: @sanskar107  it is at frame.laser_labels.num_lidar_points_in_box this should be somewhat important for detection performance so there should be some filtering here or later

**Pull request merged**